### ISSUE TITLE
Linux GUI apps aren't showing up in Start menu (shell script support in start menu shortcut)

### ISSUE THREAD

**Issue by stevenbrix** [microsoft/wslg#4]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.21343.1010]
Your Distribution version: Ubuntu 20.04
Your WSLg version: 0.2.13


```

# Steps to reproduce

Commands I ran to install Jet Brains Rider:
```
curl https://download-cf.jetbrains.com/rider/JetBrains.Rider-2020.3.4.tar.gz -o /tmp/rider.tar.gz
sudo tar -xzf /tmp/rider.tar.gz -C /opt
cd /opt/JetBrains\ Rider-2020.3.4/bin
bash ./rider.sh
```

**WSL logs**: 
[pulseaudio.log](https://github.com/microsoft/WSLG/files/6214021/pulseaudio.log)
[weston.log](https://github.com/microsoft/WSLG/files/6214024/weston.log)
[versions.txt](https://github.com/microsoft/WSLG/files/6214669/versions.txt)

#  Expected behavior

I would see the Jet Brains Rider app under the `Ubuntu` folder in the Start Menu

# Actual behavior

There is no Ubuntu folder in the Star Menu

**Comment by iamwwc** [comment]
If clion don't create shortcut for you, you can create it manually

create a file at `/usr/share/applications/clion.desktop`

change clion path for you own

```
[Desktop Entry]
Version=13.0
Type=Application
Terminal=false
Icon[en_US]=/root/clion-2021.1/bin/clion.png
Name[en_US]=Clion
Exec=/root/clion-2021.1/bin/clion.sh
Name=Clion
Icon=/root/clion-2021.1/bin/clion.png
```

**Comment by stevenbrix** [comment]
@spronovo could you share what the rider desktop file looks like? i'll try @iamwwc's suggestion and see if i can copy it to the global location

**Comment by spronovo** [comment]
> @spronovo could you share what the rider desktop file looks like? i'll try @iamwwc's suggestion and see if i can copy it to the global location

Here is the content of the file

```
[Desktop Entry]
Version=1.0
Type=Application
Name=JetBrains Rider
Icon=/opt/JetBrains Rider-2020.3.4/bin/rider.svg
Exec="/opt/JetBrains Rider-2020.3.4/bin/rider.sh" %f
Comment=A cross-platform IDE for .NET
Categories=Development;IDE;
Terminal=false
StartupWMClass=jetbrains-rider
StartupNotify=true
```

**Comment by stevenbrix** [comment]
thanks @spronovo. so even after adding that file, i still don't see anything in my start menu

**Comment by stevenbrix** [comment]
for the record, i have other `.desktop` files in that directory, but none of them show up in my start menu

**Comment by hideyukn88** [comment]
by looking at weston.log previously provided from you, it looks those .desktop have either "terminal" or "NoDisplay" property, thus those applications won't be registered up at start menu.

and if you created .desktop at /usr/share/applications, WSLg monitors that folder and dynamically adds programs start menu without reboot/restarting WSL, and if you can take a peek /mnt/wslg/weston.log, those are logged there, and if any error, it most likely logged, too.

**Comment by stevenbrix** [comment]
Hmm, it's still not launching. Something went wrong, but my weston.log file is now almost 3GB in space, so i can't open it in any editors to see what's inside
![WhenYou'reHangingOutAndRememberYouHadAnApptGIF](https://user-images.githubusercontent.com/9649518/116592897-615e4300-a8d5-11eb-995c-99ba274177ed.gif)

**Comment by hideyukn88** [comment]
I noticed "Exec="/opt/JetBrains Rider-2020.3.4/bin/rider.sh" %f" so it's shell script, I don't think this is currently possible.

**Comment by stevenbrix** [comment]
Thanks @hideyukn88, is there a known issue tracking this?

**Comment by hideyukn88** [comment]
It's not directly tracking but https://github.com/microsoft/wslg/issues/66 is the suggesting to launch start menu program under the possible choice of shell (such as bash), and we are currently looking into that, so it's possible to address this as well. We can keep this issue opened until we are sure that can provide solution when Exec is pointing shell script.

**Comment by stevenbrix** [comment]
Sounds good, i'm also starting to see these errors when i run the rider script locally, 

stevenki@poseidon-devbook:~/code/avs$ bash /opt/JetBrains\ Rider-2020.3.4/bin/rider.sh
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.

(process:25172): dconf-CRITICAL **: 11:13:51.844: failed to allocate file '/mnt/wslg/runtime-dir/dconf/user': No space left on device.  dconf will not work properly.

(process:25172): dconf-CRITICAL **: 11:13:51.847: failed to allocate file '/mnt/wslg/runtime-dir/dconf/user': No space left on device.  dconf will not work properly.

(process:25172): dconf-CRITICAL **: 11:13:51.852: failed to allocate file '/mnt/wslg/runtime-dir/dconf/user': No space left on device.  dconf will not work properly.

(process:25172): dconf-CRITICAL **: 11:13:51.854: failed to allocate file '/mnt/wslg/runtime-dir/dconf/user': No space left on device.  dconf will not work properly.

(process:25172): dconf-CRITICAL **: 11:13:51.876: failed to allocate file '/mnt/wslg/runtime-dir/dconf/user': No space left on device.  dconf will not work properly.
IDE has not been initialized yet

**Comment by hideyukn88** [comment]
Given your weston.log is 3GB, can you try to shutdown wsl (by wsl --shutdown or --terminate [distro name]). We will also trim the amount of logging or make it circular somehow.

**Comment by stevenbrix** [comment]
@hideyukn88 with big help from @craigloewen-msft this is now working! The key is to set the `StartupNotify=false`.

Below is the jetbrains-rider.desktop file I have, the only change from what @spronovo gave me is the last line.

```
[Desktop Entry]
Version=1.0
Type=Application
Name=JetBrains Rider
Icon=/opt/JetBrains Rider-2020.3.4/bin/rider.svg
Exec="/opt/JetBrains Rider-2020.3.4/bin/rider.sh" %f
Comment=A cross-platform IDE for .NET
Categories=Development;IDE;
Terminal=false
StartupWMClass=jetbrains-rider
StartupNotify=false
```

**Comment by stevenbrix** [comment]
@hideyukn88 interesting, this is now working. i wonder if my wsl was in a bad state or something

**Comment by stevenbrix** [comment]
Just for an update, i'm now back in this invalid state. I'll try restarting my machine and see if that fixes things. Here are the logs:(weston.log is empty)

[pulseaudio.log](https://github.com/microsoft/wslg/files/6422638/pulseaudio.log)
[versions.txt](https://github.com/microsoft/wslg/files/6422639/versions.txt)

Here's output from the debug console:
[wsl_output.txt](https://github.com/microsoft/wslg/files/6422643/wsl_output.txt)

**Comment by stevenbrix** [comment]
A restart does fix things, and thankfully Windows has made restarts less disruptive, but it would be nice to not require one

**Comment by spronovo** [comment]
The issue with the GUI application not showing up in the start menu in this case has been fix with today's release of 1.0.22 (weston.log should also be growing at a much slower state now :-)). To update to 1.0.22 please run the following from an elevated command prompt:

```
wsl --shutdown
wsl --update
```
